XivAlexander/App_Network_SocketHook.cpp

```diff
@@ -222,46 +224,115 @@ class App::Network::SingleConnection::Internals {
 			Misc::Logger::GetLogger().Format<LogLevel::Warning>(LogCategory::SocketHook, "%p: WSAIoctl SIO_TCP_INFO v0 failed: %08x", m_socket, WSAGetLastError());
 			m_nIoctlTcpInfoFailureCount++;
 			return 0;
-		} else if (cb != sizeof info) {
+		}
+		else if (cb != sizeof info) {
 			Misc::Logger::GetLogger().Format<LogLevel::Warning>(LogCategory::SocketHook, "%p: WSAIoctl SIO_TCP_INFO v0: buffer size mismatch (%d != %d)", m_socket, cb, sizeof info);
 			m_nIoctlTcpInfoFailureCount++;
 			return 0;
-		} else {
+		}
+		else {
 			return std::max(1LL, info.RttUs / 1000LL);
 		}
 	}
 
+	void AddConnectionLatencyItem(int64_t latency) {
+		const int latencyTrackCount = 10;
+
+		ObservedConnectionLatencyList.push_back(latency);
+
+		if (ObservedConnectionLatencyList.size() > latencyTrackCount)
+			ObservedConnectionLatencyList.erase(ObservedConnectionLatencyList.begin());
+	}
+
+	int64_t GetMedianConnectionLatency() const {
+		if (ObservedConnectionLatencyList.empty())
+			return 0;
+
+		size_t count = ObservedConnectionLatencyList.size();
+		std::vector<int64_t> SortedConnectionLatencyList(count);
+		std::partial_sort_copy(ObservedConnectionLatencyList.begin(), ObservedConnectionLatencyList.end(), SortedConnectionLatencyList.begin(), SortedConnectionLatencyList.end());
+
+		if ((count % 2) == 0)
+			// even
+			return (SortedConnectionLatencyList[count / 2] + SortedConnectionLatencyList[count / 2 - 1]) / 2;
+
+		// odd
+		return SortedConnectionLatencyList[count / 2];
+	}
+
+	int64_t GetMeanConnectionLatency() const {
+		if (ObservedServerResponseList.empty())
```

Comment:
Probably `ObservedConnectionLatencyList`?

Reply:
Oops, something must have gotten messed up in my merges